src/backend/executor/nodeShareInputScan.c

```diff
@@ -412,11 +412,12 @@ ExecReScanShareInputScan(ShareInputScanState *node)
 #include "sys/types.h"
 #include "sys/stat.h"
 #include "storage/fd.h"
+#include "storage/proc.h"
 #include "cdb/cdbselect.h"
 
 char *shareinput_create_bufname_prefix(int share_id)
 {
-	return psprintf("SIRW_%d_%d_%d", gp_session_id, gp_command_count, share_id);
+	return psprintf("SIRW_%d_%d_%d", gp_session_id, MyProc->queryCommandId, share_id);
```

Comment:
Could you please add tests for ShareInputScan?

Reply:
Problem with ShareInputScan is reproduced only together with the monitoring extension. So the test will be added to the extension code in a separate task.